caffe2/core/tensor.cc

```diff
@@ -124,6 +124,34 @@ Tensor empty(at::IntList dims, at::TensorOptions options) {
   return tensor;
 }
 
+void reinitialize_(
+    Tensor* tensor,
+    at::IntList dims,
+    at::TensorOptions options) {
+  CAFFE_ENFORCE(options.device_opt() != c10::nullopt);
+  if (*tensor) {
+    if (tensor->GetDevice() == options.device()) {
+      if (tensor->sizes() != dims) {
+        // Resize when the dims doesn't match
+        tensor->Resize(dims);
+      }
+      auto type_meta = at::scalarTypeToTypeMeta(options.dtype());
+      if (tensor->meta() == type_meta) {
+        tensor->raw_mutable_data();
+      } else {
+        // create a new Tensor when the data_type doesn't match
```

Comment:
These dtype changes are going to keep giving me nightmares.

Reply:
we could start throwing a WARNING here if the type was initialized previously and see how often it gets triggered

Reply:
For current codemod, the pattern we are matching is Tensor in memeber variable and we'll replace the first Resize call with initialize, I don't think it will get triggered. I think the question is whether we want to use this API for other purposes as well, or just for the single purpose of reinitializing a member variable of type Tensor.